pkg/config/config.go

```diff
@@ -1360,6 +1360,11 @@ func load(config Config, origin string, loadSecret bool) (*Warnings, error) {
 		applyOverrideFuncs(config)
 	}()
 
+	if IsServerless() {
+		log.Debug("No config file detected, using environment variable based configuration")
+		return &warnings, nil
+	}
+
 	if err := config.ReadInConfig(); err != nil {
```

Comment:
The debug log seems a little bit misleading compared to the logic of the current version of the code:
In serverless environment, we’ll log `No config file detected` before checking if there’s a config file or not.

I understand that agent config file are currently not needed in serverless environment but, if some day a customer imagines a use case for such a setup, it would be clearer to log something like this:

```suggestion
	if IsServerless() {
		log.Debug("Running on a serverless environment. Ignoring config file. Using environment variable based configuration only")
		return &warnings, nil
	}

	if err := config.ReadInConfig(); err != nil {
```